ruleset/rules/0910-ms-exchange-proxylogon_rules.xml

```diff
@@ -2,132 +2,127 @@
-        <description>Possible Nishang Invoke-PowerShellTcpOneLine reverse shell script execution</description>
-        <mitre>
-        <id>T1210</id>
-        <id>T1588</id>
-        <id>T1059</id>
-        </mitre>
-    </rule>
+  <rule id="91000" level="12">
+    <if_group>sysmon_event1</if_group>
+    <field name="win.eventdata.parentImage">\\UMWorkerProcess.exe</field>
+    <field name="win.eventdata.originalFileName" negate="yes">werfault.exe|wermgr.exe</field>
+    <description>UMWorkerProcess.exe spawned unexpected processes. Possible malicious activity.</description>
+    <mitre>
+      <id>T1210</id>
+    </mitre>
+  </rule>
+
+  <rule id="91001" level="12">
+    <if_group>sysmon_event_11</if_group>
+    <field name="win.eventdata.image">\\UMWorkerProcess.exe</field>
+    <field name="win.eventdata.targetFilename" negate="yes">CacheCleanup.bin$|.txt$|.log$|.cfg$|cleanup.bin$</field>
+    <description>UMWorkerProcess.exe spawned creating abnormal files. Possible malicious activity.</description>
+    <mitre>
+      <id>T1210</id>
+    </mitre>
+  </rule>
+
+  <!--
+    CVE-2021-26855 (ProxyLogon) exploitation can be detected via the following Exchange HttpProxy logs:
+    These logs are located in the following directory: %PROGRAMFILES%\Microsoft\Exchange Server\V15\Logging\HttpProxy
+    Exploitation can be identified by searching for log entries where the AuthenticatedUser is empty and the AnchorMailbox contains the pattern of ServerInfo~*/*
+  -->
+  <rule id="91002" level="12">
+    <decoded_as>HTTPProxyLog</decoded_as>
+    <regex type="pcre2">,,[^,]*,ServerInfo~[^,]*/[^,]*</regex>
+    <description>MS Exchange - Possible ProxyLogon vulnerability exploitation (CVE-2021-26855).</description>
+    <mitre>
+      <id>T1210</id>
+    </mitre>
+  </rule>
+
+  <!--
+    CVE-2021-27065  is a post-authentication arbitrary file write vulnerability in Exchange. If HAFNIUM could authenticate
+    with the Exchange server then they could use this vulnerability to write a file to any path on the server.
+    They could authenticate by exploiting the CVE-2021-26855 SSRF vulnerability or by compromising a legitimate
+    admin’s credentials.
+  -->
+  <rule id="91003" level="12">
+    <decoded_as>ECPServerLog</decoded_as>
+    <regex type="pcre2">Set-.+VirtualDirectory.+?Url.+\<\w+.*\>.*?\<\/\w+\>.+?VirtualDirectory</regex>
+    <description>MS Exchange - Possible ProxyLogon vulnerability exploitation (CVE-2021-27065).</description>
+    <mitre>
+      <id>T1210</id>
+    </mitre>
+  </rule>
+
+  <!--
+    CVE-2021-26858 exploitation can be detected via the Exchange log files:
+    C:\Program Files\Microsoft\Exchange Server\V15\Logging\OABGeneratorLog
+    Files should only be downloaded to the %PROGRAMFILES%\Microsoft\Exchange Server\V15\ClientAccess\OAB\Temp directory.
+    In case of exploitation, files are downloaded to other directories (UNC or local paths).
+  -->
```

Comment:
Criteria say there shouldn't be logs in here, but they should be in ini instead. Since windows doesn't have ini I'm not sure if this is fine.